src/components/housing/carrousel.tsx

```diff
@@ -1,43 +1,47 @@
-import React, { useState } from "react"
-
+import React from "react"
+/**
+ * Just checking if the props are valids
+ */
+ interface Props {
+    pictures: Array<string>,
+}
 /**
  * Hero of Housing page, display the housing pictures. Will switch between pictures 
  * when the user clicks on the next or previous button
  */
-export default function Carrousel(props:Props):JSX.Element {
-    const [slide, setSlide] = useState(0);
-    const slides:number = props.pictures.length
-    return (
-        <section className="relative">
+ export default class Carrousel extends React.Component<any, any> {
+    constructor(props:Props) {
+        super(props)
+        this.state = {
+            slide: 0,
+            slides: props.pictures.length
+        }
+    }
+    render() {
+        return <section className="relative">
             <div className="flex flex-nowrap group text-center hero-index h-96 rounded-3xl w-full overflow-hidden">
-                {props.pictures.map((picture, index) => {
+                {this.props.pictures.map((picture: string | undefined, index: React.Key | null | undefined) => {
                     return(<img className="flex-none object-cover h-full w-full translate-x-0 transition-all duration-500"
-                        src={picture} key={index} style={{transform: "translateX(-" + slide*100 + "%)"}}>
+                        src={picture} key={index} style={{transform: "translateX(-" + this.state.slide*100 + "%)"}}>
                     </img>)
                 })}
             </div>
-            { slides === 1 ? "" : <>
+            { this.state.slides === 1 ? "" : <>
             <button className="text-xl absolute left-4 top-40" title="Photo précédente"
                     // Every day I wake up, then I remember that the most popular programming language in the world 
                     // can't do a simple modulo, and I am sad for the day 😔
-                    onClick={() => {setSlide(((slide-1 % slides) + slides) % slides )}}> 
+                    onClick={() => {this.setState({slide: (((this.state.slide-1 % this.state.slides) + this.state.slides) % this.state.slides )})}}> 
```

Comment:
Codacy has a fix for the issue: [Missing semicolon.](https://app.codacy.com/gh/GoulvenC/GoulvenClech_11_02082021/pullRequest?prid=8201079)
```suggestion
                    onClick={() => {this.setState({slide: (((this.state.slide-1 % this.state.slides) + this.state.slides) % this.state.slides )});}}> 
```